src/datasets/utils/py_utils.py

```diff
@@ -1349,6 +1374,13 @@ def iflatmap_unordered(
                 except Empty:
                     if all(async_result.ready() for async_result in async_results) and queue.empty():
                         break
+                if pool.has_restarted_subprocess():
+                    # One of the subprocesses has died. We should not wait forever.
+                    raise RuntimeError(
+                        "One of the subprocesses has abruptly died during map operation."
+                        "To debug the error, disable multiprocessing."
+                    )
```

Comment:
Could we do this automatically outside of `iflatmap_unordered` ? There is `pool.append()` that is automatically called in a thread to repopulate processes that crashed. Maybe you can just override `Pool.append` to raise an error if `pool_state` is set and `pool._state == multiprocessinf.pool.RUN`.

Reply:
hmm maybe the error wouldn't be handled by the thread correctly and the pool wouldn't close gracefully